data/core/common.lua

```diff
@@ -304,19 +446,30 @@ function common.serialize(val, opts)
 end
 
 
+---Returns the last portion of a path.
+---If this function doesn't have a filename (eg. `"C:/"` or `"/"`),
+---this function may return nil.
```

Comment:
Can this actually happen? I think this just returns the original path in that case.